### DISCUSSION TITLE
AUTH_FAILED issues on VyprVPN, out of ideas!

### DISCUSSION THREAD

**Discussion by rodti** [haugene/docker-transmission-openvpn#1998] [Q&A] [answered]
I was previously using this container with SlickVPN which worked pretty well. The VPN was terrible, but the container was reliable. I used the built-in SlickVPN configs.

I've moved to VyprVPN for a more reliable VPN experience.

I initially tried changing OPENVPN_PROVIDER to VYPRVPN, and of course updating the authentication and config details. This caused endless AUTH_FAILED errors.

The openvpn-credentials.txt file is valid, readable, and has no special characters.

I thought there might be a problem with repurposing the existing container from SlickVPN to VyprVPN, so I deleted it and created a new one. This still gave me the same AUTH_FAILED errors.

I've tried using a default OPENVPN_CONFIG, a number of different ones, and a custom one with manually downloaded certificate and .ovpn file. All give AUTH_FAILED errors.

I have reset my password with VyprVPN, waited 24 hours, and tried again. AUTH_FAILED errors.

The VPN account works from my IP address on multiple machines using various VPN protocols, and on my main machine using VyprVPN's app (configured to use OpenVPN).

I'm absolutely stumped!

Here's a log of a typical error:

```
Starting container with revision: ae45b1f22178ddee089d9ea10d4732dbde2d8ef2
Creating TUN device /dev/net/tun
Using OpenVPN provider: VYPRVPN
Running with VPN_CONFIG_SOURCE auto
Provider VYPRVPN has a bundled setup script. Defaulting to internal config
Executing setup script for VYPRVPN
ovpn files detected, not downloading configs
Starting OpenVPN using config United Kingdom-256.ovpn
Modifying /etc/openvpn/vyprvpn/United Kingdom-256.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Setting OpenVPN credentials...
Fri Oct  8 15:35:26 2021 OpenVPN 2.4.7 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 19 2021
Fri Oct  8 15:35:26 2021 library versions: OpenSSL 1.1.1f  31 Mar 2020, LZO 2.10
Fri Oct  8 15:35:26 2021 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
Fri Oct  8 15:35:26 2021 TCP/UDP: Preserving recently used remote address: [AF_INET]209.99.22.18:443
Fri Oct  8 15:35:26 2021 Socket Buffers: R=[212992->212992] S=[212992->212992]
Fri Oct  8 15:35:26 2021 UDP link local: (not bound)
Fri Oct  8 15:35:26 2021 UDP link remote: [AF_INET]209.99.22.18:443
Fri Oct  8 15:35:26 2021 TLS: Initial packet from [AF_INET]209.99.22.18:443, sid=6993d949 cc056bf2
Fri Oct  8 15:35:26 2021 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
Fri Oct  8 15:35:27 2021 VERIFY OK: depth=2, C=CH, ST=Lucerne, L=Meggen, O=Golden Frog GmbH, CN=Golden Frog GmbH Root CA, emailAddress=admin@goldenfrog.com
Fri Oct  8 15:35:27 2021 VERIFY OK: depth=1, C=CH, ST=Lucerne, L=Meggen, O=Golden Frog GmbH, CN=VyprVPN Intermediate CA, emailAddress=admin@goldenfrog.com
Fri Oct  8 15:35:27 2021 VERIFY X509NAME OK: C=CH, ST=Lucerne, L=Meggen, O=Golden Frog GmbH, CN=uk1.vyprvpn.com, emailAddress=admin@goldenfrog.com
Fri Oct  8 15:35:27 2021 VERIFY OK: depth=0, C=CH, ST=Lucerne, L=Meggen, O=Golden Frog GmbH, CN=uk1.vyprvpn.com, emailAddress=admin@goldenfrog.com
Fri Oct  8 15:35:29 2021 Control Channel: TLSv1.2, cipher TLSv1.2 ECDHE-RSA-AES256-GCM-SHA384, 4096 bit RSA
Fri Oct  8 15:35:29 2021 [uk1.vyprvpn.com] Peer Connection Initiated with [AF_INET]209.99.22.18:443
Fri Oct  8 15:35:30 2021 SENT CONTROL [uk1.vyprvpn.com]: 'PUSH_REQUEST' (status=1)
Fri Oct  8 15:35:30 2021 AUTH: Received control message: AUTH_FAILED
Fri Oct  8 15:35:30 2021 SIGTERM[soft,auth-failure] received, process exiting
```

Has anyone else had similar issues with this container and VyprVPN? It's really frustrating as this container has usually been absolutely dependable and I can't see what's causing the issue.

**Comment by rodti** [comment]
I can confirm I also have the same problem with v3.7.1 of the container.

**Comment by pkishino** [comment]
I’d contact vypervpn support and verify those ovpn configs.. sounds like they might be outdated or they changed stuff on server end to require higher version of openvpn?

**Reply by rodti** [reply]
Yeah will do, odd that I've got the latest ones from them to try a custom config, will see what they say.

**Comment by rodti** [comment, marked as answer]
Okay, this is really embarrassing, but I've figured it out. I've taken out a Giganews Usenet subscription, and with that subscription I get the VyprVPN service. I've set up the container with VYPRVPN as the provider, checked that VyprVPN uses my standard Giganews login details, and as you can see I've done lots of troubleshooting.

I was going through the VPN config repo and I saw there was a GIGANEWS provider config. I've switched to that and it worked first time.

You live and learn, eh?
